cub/cub/device/dispatch/dispatch_transform.cuh

```diff
@@ -316,28 +316,38 @@ struct dispatch_t<StableAddress,
       return config.error();
     }
 
-    auto loaded_bytes_per_iter = kernel_source.LoadedBytesPerIteration();
-    // choose items per thread to reach minimum bytes in flight
-    const int items_per_thread =
-      loaded_bytes_per_iter == 0
-        ? +policy.ItemsPerThreadNoInput()
-        : ::cuda::ceil_div(policy.min_bif, config->max_occupancy * block_dim * loaded_bytes_per_iter);
-
-    // but also generate enough blocks for full occupancy to optimize small problem sizes, e.g., 2^16 or 2^20 elements
-    const int items_per_thread_evenly_spread = static_cast<int>(
-      (::cuda::std::min)(Offset{items_per_thread}, num_items / (config->sm_count * block_dim * config->max_occupancy)));
-
-    const int items_per_thread_clamped =
-      ::cuda::std::clamp(items_per_thread_evenly_spread, +policy.MinItemsPerThread(), +policy.MaxItemsPerThread());
-    const int tile_size = block_dim * items_per_thread_clamped;
+    const int ipt = [&] {
```

Comment:
I am slightly concerned about the use of immediately invoked lambdas in this function. This function is variadic, so we will generate a bunch of types for every instantiation

Reply:
I can pull out a utility function instead, but that will lead to similar instantiations. Otherwise:
```c++
int ipt;
do {
      if constexpr (Algorithm::vectorized == ActivePolicy::algorithm)
      {
        if (can_vectorize)
        {
          ipt = wrapped_policy.ItemsPerThreadForVectorizedPath();
          break;
        }
      }
      .. // long fallback
      ipt = ...;
}
while(0):
```
or a `goto` does not loog great either. What's your recommendation?

Reply:
@miscco any strong opposition against the lambda? I would like to keep it.

Reply:
Ping.

Reply:
No change requested